<launch>
    <!-- Planning  parameters-->
    <arg name="map_size_x_" value="30.0"/>
    <arg name="map_size_y_" value="30.0"/>
    <arg name="map_size_z_" value=" 3.0"/>
    

  <node name="formation_planning" pkg="bspline_race" type="formation_planning" output="screen">
        <!-- Plan Manager -->
        <param name="gvf/cloud_topic"      type="string" value="/mock_map" />
        <param name="gvf/odom_topic"       type="string" value="/sim/odom" />
        <param name="gvf/planInterval"     value="0.2" /> 
        <!-- SDF Map -->
        <param name="sdf_map/resolution"   value="0.1" /> 
        <param name="sdf_map/map_size_x"   value="$(arg map_size_x_)" /> 
        <param name="sdf_map/map_size_y"   value="$(arg map_size_y_)" /> 
        <param name="sdf_map/map_size_z"   value="$(arg map_size_z_)" /> 

        <param name="sdf_map/local_update_range_x"  value="8.0" /> 
        <param name="sdf_map/local_update_range_y"  value="8.0" /> 
        <param name="sdf_map/local_update_range_z"  value="3.0" /> 
        
        <param name="sdf_map/obstacles_inflation"     value="0.099" /> 
        <param name="sdf_map/local_bound_inflate"    value="0.0"/>
        <param name="sdf_map/local_map_margin" value="50"/>
        <param name="sdf_map/ground_height"        value="-0.01"/>

        <!-- depth filter -->
        <param name="sdf_map/use_depth_filter" value="true"/>
        <param name="sdf_map/depth_filter_tolerance" value="0.15"/>
        <param name="sdf_map/depth_filter_maxdist"   value="5.0"/>
        <param name="sdf_map/depth_filter_mindist"   value="0.2"/>
        <param name="sdf_map/depth_filter_margin"    value="2"/>
        <param name="sdf_map/k_depth_scaling_factor" value="1000.0"/>
        <param name="sdf_map/skip_pixel" value="2"/>
        <!-- local fusion -->
        <param name="sdf_map/p_hit"  value="0.65"/>
        <param name="sdf_map/p_miss" value="0.35"/>
        <param name="sdf_map/p_min"  value="0.12"/>
        <param name="sdf_map/p_max"  value="0.90"/>
        <param name="sdf_map/p_occ"  value="0.80"/>
        <param name="sdf_map/min_ray_length" value="0.5"/>
        <param name="sdf_map/max_ray_length" value="4.5"/>
        <param name="sdf_map/esdf_slice_height" value="0.3"/>
        <param name="sdf_map/visualization_truncate_height"   value="2.49"/>
        <param name="sdf_map/virtual_ceil_height"   value="2.5"/>
        <param name="sdf_map/show_occ_time"  value="false"/>
        <param name="sdf_map/show_esdf_time" value="false"/>
        <param name="sdf_map/pose_type"     value="2"/>  
        <param name="sdf_map/frame_id"      value="world"/>

        <!-- geometric path searching -->
        <param name="astar/lambda_heu" value="10.0" type="double"/>
        <param name="astar/resolution_astar" value="0.1" type="double"/>
        <param name="astar/time_resolution" value="0.8" type="double"/>
        <param name="astar/margin" value="0.1" type="double"/>
        <param name="astar/allocate_num" value="1000000" type="int"/>

    </node>


</launch>